packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -3,21 +3,164 @@ import {
 import { CloudControlClient, GetResourceCommand } from '@aws-sdk/client-cloudcontrol';
+import { AppSyncClient, GetDataSourceCommand } from '@aws-sdk/client-appsync';
 import { unset } from 'lodash';
+import { createHash } from 'crypto';
+import { userInfo } from 'os';
 
+type AppId = string;
+type ProjectName = string;
+type BranchName = string;
+type SandboxName = string;
+
+type BackendIdentifier =
+  | {
+      namespace: Readonly<AppId>;
+      name: Readonly<BranchName>;
+      type: Readonly<'branch'>;
+      hash?: Readonly<string>;
+    }
+  | {
+      namespace: Readonly<ProjectName>;
+      name: Readonly<SandboxName>;
+      type: Readonly<'sandbox'>;
+      hash?: Readonly<string>;
+    };
+
+const STACK_NAME_LENGTH_LIMIT = 128;
+const AMPLIFY_PREFIX = 'amplify';
+const HASH_LENGTH = 10;
+const NUM_DASHES = 4;
 const pushTimeoutMS = 1000 * 60 * 20; // 20 minutes;
 
+function toStackName(backendId: BackendIdentifier): string {
+  const hash = getHash(backendId);
+
+  // only take the first 50 chars here to make sure there is room in the stack name for the namespace as well
+  const name = sanitizeChars(backendId.name).slice(0, 50);
+
+  const namespaceMaxLength =
+    STACK_NAME_LENGTH_LIMIT - AMPLIFY_PREFIX.length - backendId.type.length - name.length - NUM_DASHES - HASH_LENGTH;
+
+  const namespace = sanitizeChars(backendId.namespace).slice(0, namespaceMaxLength - 1);
+
+  return ['amplify', namespace, name, backendId.type, hash].join('-');
+}
+
+const getHash = (backendId: BackendIdentifier): string =>
+  backendId.hash ??
+  // md5 would be sufficient here because this hash does not need to be cryptographically secure, but this ensures that we don't get unnecessarily flagged by some security scanner
+  createHash('sha512').update(backendId.namespace).update(backendId.name).digest('hex').slice(0, HASH_LENGTH);
+
+/**
+ * Remove all non-alphanumeric characters from the input string
+ */
+const sanitizeChars = (str: string): string => {
+  return str.replace(/[^A-Za-z0-9]/g, '');
+};
+
+export async function copyFunctionFile(projRoot: string, gen1FunctionName: string): Promise<void> {
+  const sourcePath = path.join(
+    projRoot,
+    '.amplify',
+    'migration',
+    'amplify',
+    'backend',
+    'function',
+    gen1FunctionName.split('-')[0],
+    'src',
+    'index.js',
+  );
+  const destinationPath = path.join(projRoot, 'amplify', 'function', gen1FunctionName.split('-')[0], 'handler.ts');
+  const content = await fs.readFile(sourcePath, 'utf8');
+
+  // Replace the first occurrence of 'event' with 'event: any'
+  const modifiedContent = content.replace(/(exports\.handler\s*=\s*async\s*\(\s*)event(\s*\))/, '$1event: any$2');
+
+  await fs.writeFile(destinationPath, modifiedContent, 'utf8');
+}
+
+export async function copyGen1Schema(projRoot: string): Promise<void> {
+  const gen1SchemaPath = path.join(projRoot, '.amplify', 'migration', 'amplify', 'backend', 'api', 'codegentest', 'schema.graphql');
```

Comment:
`codegentest` -> make it a constant?

Reply:
Addressed. https://github.com/aws-amplify/amplify-cli/pull/13952/commits/d310c32e939e79bfc8ab2ae81be8d17123a4072f